DirectX11Proj/graphicsclass.cpp

```diff
@@ -136,14 +135,19 @@ bool GraphicsClass::Render(IScene *const aScene)
 	{
 		//std::cout << i << std::endl;
 		aScene->mObjects[i]->mpModel->Render(mpDirect3D->GetDeviceContext());
-		result = m_ColorShader->Render(mpDirect3D->GetDeviceContext(), aScene->mObjects[i]->mpModel->GetIndexCount(), worldMatrix, viewMatrix, projectionMatrix);
-	}
-	// Render the model using the color shader.
-	if (!result)
-	{
-		return false;
+
+		if (aScene->mObjects[i]->mpModel->mHastexture)
```

Comment:
Is an object a class? Call GetModel() and HasTexture() (or GetTextureCount()) instead. Also check for nullptrs